graphql/handler/transport/headers.go

```diff
@@ -1,10 +1,51 @@
 package transport
 
-import "net/http"
+import (
+	"mime"
+	"net/http"
+	"strings"
+)
+
+const (
+	acceptApplicationJson                = "application/json"
+	acceptApplicationGraphqlResponseJson = "application/graphql-response+json"
+)
+
+func determineResponseContentType(explicitHeaders map[string][]string, r *http.Request) string {
+	for k, v := range explicitHeaders {
+		if strings.EqualFold(k, "Content-Type") {
+			return v[0]
+		}
+	}
+
+	accept := r.Header.Get("Accept")
+	if accept == "" {
+		return acceptApplicationGraphqlResponseJson
+	}
+
+	for _, acceptPart := range strings.Split(accept, ",") {
+		mediaType, _, err := mime.ParseMediaType(strings.TrimSpace(acceptPart))
+		if err != nil {
+			continue
+		}
+		switch mediaType {
+		case "*/*", "application/*":
+			return acceptApplicationGraphqlResponseJson
+		case "application/json":
+			return acceptApplicationJson
+		case "application/graphql-response+json":
+			return acceptApplicationGraphqlResponseJson
+		}
+	}
+
+	return acceptApplicationGraphqlResponseJson
+}
 
 func writeHeaders(w http.ResponseWriter, headers map[string][]string) {
 	if len(headers) == 0 {
 		headers = map[string][]string{
+			// Stay with application/json (not application/graphql-response+json)
+			// as it is not an actively supported protocol for now
```

Comment:
I'm wondering about this part and Apollo's GraphQL Federation, but I think it should be ok.